apps/tasks/models.py

```diff
@@ -1,45 +1,104 @@
 import datetime
 
+from django.core.mail import send_mail
 from django.db import models
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
 from django.utils import timezone
+from rest_framework.exceptions import ValidationError
 
 from apps.accounts.models import User
 
 
 # Create your models here.
 
+class TaskQuerySet(models.QuerySet):
+    def with_total_duration(self):
+        return self.annotate(total_duration=(models.Sum('timelog_task__duration')))
+
+
 class Task(models.Model):
+    objects = TaskQuerySet.as_manager()
 
     class Status(models.TextChoices):
-        todo = 'to_do', 'To do'
-        assigned = 'assigned', 'Assigned'
-        in_progress = 'in_progress', 'In progress'
-        completed = 'completed', 'Completed'
+        TODO = 'to_do'
+        IN_PROGRESS = 'in_progress'
+        COMPLETED = 'completed'
 
     title = models.CharField(max_length=255)
-    description = models.CharField(max_length=255, blank=True)
-    status = models.CharField(max_length=16, choices=Status.choices, default=Status.todo)
-    assigned_to = models.ForeignKey(User, on_delete=models.CASCADE, related_name='owner', null=True)
-    created_by = models.ForeignKey(User, on_delete=models.CASCADE, related_name='created_by', null=True)
-
-    def __str__(self):
-        return f" \"{self.title}\" By: {self.assigned_to.get_full_name()}"
+    description = models.TextField(blank=True)
+    status = models.CharField(max_length=50, choices=Status.choices, default=Status.TODO)
+    assigned_to = models.ForeignKey(User, on_delete=models.CASCADE, related_name='assigns', null=True)
+    created_by = models.ForeignKey(User, on_delete=models.CASCADE, related_name='created_tasks', null=True)
 
 
 class Comment(models.Model):
     task = models.ForeignKey(Task, on_delete=models.CASCADE)
-    text = models.CharField(max_length=255)
+    text = models.TextField()
 
     def __str__(self):
         return f'{self.task.title} : {self.text}'
 
 
-class Timelog(models.Model):
-    task = models.ForeignKey(Task, on_delete=models.CASCADE)
-    start = models.DateTimeField(default=timezone.now)
-    duration = models.IntegerField(default=None, null=True)
+class TimeLog(models.Model):
+    task = models.ForeignKey(Task, on_delete=models.CASCADE, related_name='timelog_task', null=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='timelog_user', null=True)
+    started_at = models.DateTimeField(default=timezone.now)
+    duration = models.DurationField(default=datetime.timedelta(), null=True)
```

Comment:
Pentru ce e default=datetime.timedelta() ?

Reply:
Trebuie de vazut daca se poate fara modelul Timer

De exemplu:
- started_at va fi null=True
- if not TimeLog.started_at inseamna ca nu e pornit
- faci metoda start care atribuie timezone.now()
- faci metoda stop care calculeaza duration, TimeLog.started_at - timezone.now()